### ISSUE TITLE
Medbay cryo pods

### ISSUE THREAD

**Issue by Carbonhell** [space-wizards/space-station-14#11245]
I'm already developing them, I have sorted out both the atmos part and the chemistry part, I need to get the actual sprite and decide whether to make an UI for it and what said UI should show.

# Summary
Cryo pods (also called cryo tubes) are specialized medical machines which allow medics to safely drop the temperature of a patient to low values, enough for chemicals such as Cryoxadone and Clonexadone to take effect.
The machine needs a considerable setup involving several atmos devices to actually do its job. There are also several ways to fuck up. While tossing a patient in a cryo pod will heal pretty much any damage without intervention, one must be careful the atmos setup isn't tampered with, along with refilling the beaker with the correct chems and so on.

# Atmos setup
(I'll post an example screenshot later)
An example atmos setup involves:
1) A connector port.
2) A canister of the right type based on the patient we are healing (oxygen / nitrogen, I think maybe an air canister works on both races?).
3) A pump to cap the output of the connector port to an acceptable pressure for patients (one atmosphere).
4) A freezer to cool the gas enough.
5) Pipes to connect everything.

I still need to figure out whether in the time of a round, the gases exhaled by patients require to be scrubbed out or not. If they do, a portable scrubber and an additional connector port might be enough.

# Interaction
As for now I have implemented verbs to interact with the pod. One can put a beaker in by clicking the pod with it. A mob can be put in the pod by dragging (or being dragged) into it. Ejection occurs by right clicking the pod, choosing eject and choosing whether to eject the mob or the beaker.

I personally think patients should be able to eject themself from the tube (maybe patient ejection might have a slight timer of 2-3 seconds instead of being instant, whatever). I have several bad experiences of being left in a pod to rot on ss13 when self ejection was disabled. This would make tampering with pods harder though, since the victim can just press eject and seek help.

# UI
I have not coded this part yet, and it is the part where I need feedback in particular. The verbs interaction seems fine in my opinion. The only issues right now would be to add a way to analyze the patient while inside the tube, for which an UI might help. Or I could just allow somehow to hit the pod with a health analyzer and relay the action on the patient inside.
Other than that, the UI could show if the setup is safe or not and what chemical is contained inside the beaker inserted in the pod.

# Post-treatment
Due to how the cryo pods work right now, they change your cold damage threshold to an extremely low value. This means the patient will still get the alert about extremely cold air, without actually taking damage.
This only applies while inside the pod. Once you exit, you will start taking cold damage while heating up. Without intervention, it takes around 30 seconds and with some leftover Cryoxadone in your bloodstream, you will take around 15 cold damage. I think this is a nice nerf so that you actually need medical help and the cryo pods would only be used for serious patients (>50 damage I guess?). Or you could use a mix of Cryoxadone and something else to negate the damage, etc.

# Sprites
I am using an existing placeholder for now. I plan to get it from /tg/ if the license allows for it. If anyone else has a better one, tell me!

# Mapping
I think it might be interesting to not map cryo pods in the maps. It would give medics a reason to call atmos techs for help in setting it up, or even find other interesting setups I haven't thought about (or a single setup for multiple tubes, you get the idea). I think the best choice would be to lock cryo pod circuit boards behind science, perhaps also spawn one or two in the tech secure storage with the other boards.
The bigger maps could maybe have one at most spawned already in, with a basic setup like the one I described. My PR will not map them in anyhow for now.

TL;DR i need to figure out if an UI would be good or not and whether the cold side effects can be left in as an interesting nerf.
I'll make a wiki page once the PR is ready.

**Comment by HoidC** [comment]
You could make ejection something toggleable via screwdriver panel?

**Comment by Carbonhell** [comment]
@HoidC Yeah, it could be an interesting mechanic to enable/disable locking by hacking the pod, I'll try to implement it.
Anyway here's a pic of a possible, working atmos setup:
![image](https://user-images.githubusercontent.com/11891037/190105649-41e8f40a-b294-47b8-ab46-6384151e5559.png)
I think I'll be able to open up a PR by the end of this week.
